src/main/java/org/openstreetmap/atlas/geography/atlas/statistics/coverage/poi/SimpleCoverage.java

```diff
@@ -0,0 +1,380 @@
+            if (isCounted(item, group))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    protected String subType()
+    {
+        return "true";
+    }
+
+    /**
+     * @return All the tag key/value pairs that an entity needs to have to be counted. Extending
+     *         this definition, if a TagGroup is empty, then all features are valid and will be
+     *         counted.
+     */
+    protected abstract Set<TagGroup> validKeyValuePairs();
+
+    /**
+     * Gets the length of the item to count
+     *
+     * @param item
+     *            The item to count
+     * @return The length of the item to count. In case this is a {@link Relation}, the length is
+     *         the sum of all the relation's lowest order {@link LineItem}s.
+     */
+    private Distance getDistance(final AtlasEntity item)
+    {
+        Distance result = Distance.ZERO;
+        if (item instanceof LineItem)
+        {
+            return ((LineItem) item).asPolyLine().length();
+        }
+        else if (item instanceof Relation)
+        {
+            for (final RelationMember member : ((Relation) item).members())
+            {
+                result = result.add(getDistance(member.getEntity()));
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Gets the area of the item to count
+     *
+     * @param item
+     *            The item to count
+     * @return The area of the item to count. In case this is a {@link Relation}, and the
+     *         {@link Relation} is type=multipolygon, the area is the area of the multipolygon. In
+     *         case the relation is of another type, the area is the sum of all the lowest order
+     *         areas that make the relation.
+     */
+    private Surface getSurface(final AtlasEntity item)
+    {
+        Surface result = Surface.MINIMUM;
+        if (item instanceof Relation || item instanceof Area)
+        {
+            try
+            {
+                final RelationOrAreaToMultiPolygonConverter converter = new RelationOrAreaToMultiPolygonConverter();
+                result = result.add(converter.apply(item).surface());
+                return result;
+            }
+            catch (final CoreException e)
+            {
+                // Many features will not be multipolygons.
+            }
+        }
+        if (item instanceof Relation)
+        {
+            // Relation that is not of type multipolygon
+            for (final RelationMember member : ((Relation) item).members())
+            {
+                result = result.add(getSurface(member.getEntity()));
+            }
+        }
+        return result;
+    }
+
+    private boolean isCounted(final AtlasEntity item, final MultiMap<String, String> andGroup)
+    {
+        for (final String key : andGroup.keySet())
+        {
+            if (item.containsValue(key, andGroup.get(key)))
+            {
+                return true;
```

Comment:
How about using anyMatch here and below?

Reply:
So all this code comes from CountCoverage that I renamed to SimpleCoverage so I'd rather not touch...